rfd/0061-tsh-aliases.md

```diff
@@ -138,23 +168,23 @@ using `tsh connect leaf ubuntu@node-1` command:
 
 ```yaml
 aliases:
-    "connect": "bash -c '$TSH login $1 && $TSH ssh $2'"
+  "connect": "exec -- -c '$TSH login $0 && $TSH ssh $1'"
 ```
 
 The following alias will list nodes in all clusters:
 
 ```yaml
 aliases:
-    "lsall": "bash -c 'for cluster in $($TSH clusters | tail -3 | head -2 | cut -d \' \' -f1); $TSH ls --cluster=$cluster; done'"
+  "lsall": "exec -- -c 'for cluster in $($TSH clusters | tail -3 | head -2 | cut -d \' \' -f1); $TSH ls --cluster=$cluster; done'"
 ```
 
 Multiple aliases can be defined in the config and can reference each other:
 
 ```yaml
 aliases:
-    "login": "$TSH login --auth=local --user=alice $@"
-    "ssh": "$TSH ssh -P $@"
-    "connect": "bash -c '$TSH login $1 && $TSH ssh $2'"
+  "login": "login --auth=local --user=alice"
```

Comment:
The `$@` here was indicating that we should add all alias's argument. How would it know to preserve arguments otherwise?

Reply:
Do it always. This is what `git` does for example and I think is a sensible default. Pretty sure people are going to be surprised if it _doesn't_ work that way. That would mean that alias

`"ssh": "ssh -P"`

would never work as there would be no way to provide the arguments.

Ignoring the provided arguments is the weirder option, I think. It is still possible, if someone needed that, by going through `tsh exec`: 

`"ssh-no-args": "exec -- -c 'echo ignoring args $@ && $TSH ssh root@node'"`